src/webviews/extension-side/dataviewer/dataViewerCommandRegistry.ts

```diff
@@ -136,4 +123,39 @@ export class DataViewerCommandRegistry implements IExtensionSingleActivationServ
             }
         }
     }
+
+    // For the given debug adapter, return the PythonEnvironment used to launch it
+    // Mirrors the logic from Python extension here:
+    // https://github.com/microsoft/vscode-python/blob/35b813f37d1ceec547277180e3aa07bd24d86f89/src/client/debugger/extension/adapter/factory.ts#L116
+    private async getDebugAdapterPython(
```

Comment:
Is it possible to have some kind of test to ensure we don't break again,
E.g. a debugger test that verifies this part.
The problem is this isn't an API contract, hence Python extension can change this anytime & we'd break again & the only way we'd know about this breakage is when a user reports it

Reply:
Yeah, taking a peek at that now. At a minimum I could verify the debug configuration looks as expected (not sure yet on how I would best trigger the context menu command).

@DonJayamanne Questions for ya, since I don't think that I've ever added anything to it. Should I maybe just talk to the python team and just expose this in the API?

Reply:
Don't see any harm in that, though it feels like an API for this might be overkill.

Reply:
This wouldn't be a problem if the debugger was part of the python extension 😉

Reply:
Yeah, I feel like we have a few areas like that, "light" python dependencies. I added a full stack test here (up to opening the data viewer) that should help to cover this. I don't think that we had testing for any of this scenario before.